pkg/metainfo/kvmetainfo/objects_test.go

```diff
@@ -15,46 +14,105 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"storj.io/storj/internal/memory"
-	"storj.io/storj/pkg/storage/objects"
 	"storj.io/storj/pkg/storage/streams"
 	"storj.io/storj/pkg/storj"
 	"storj.io/storj/pkg/stream"
-	"storj.io/storj/storage"
 )
 
-func TestGetObject(t *testing.T) {
+const TestFile = "test-file"
+
+func TestCreateObject(t *testing.T) {
+	customRS := storj.RedundancyScheme{
+		Algorithm:      storj.ReedSolomon,
+		RequiredShares: 29,
+		RepairShares:   35,
+		OptimalShares:  80,
+		TotalShares:    95,
+		ShareSize:      2 * memory.KB.Int32(),
+	}
+
+	customES := storj.EncryptionScheme{
+		Cipher:    storj.Unencrypted,
+		BlockSize: 1 * memory.KB.Int32(),
+	}
+
 	runTest(t, func(ctx context.Context, db *DB) {
 		bucket, err := db.CreateBucket(ctx, TestBucket, nil)
 		if !assert.NoError(t, err) {
 			return
 		}
 
-		store, err := db.buckets.GetObjectStore(ctx, bucket.Name)
-		if !assert.NoError(t, err) {
-			return
+		for i, tt := range []struct {
+			create     *storj.CreateObject
+			expectedRS storj.RedundancyScheme
+			expectedES storj.EncryptionScheme
+		}{
+			{
+				create:     nil,
+				expectedRS: defaultRS,
+				expectedES: defaultES,
+			}, {
+				create:     &storj.CreateObject{RedundancyScheme: customRS, EncryptionScheme: customES},
+				expectedRS: customRS,
+				expectedES: customES,
+			}, {
+				create:     &storj.CreateObject{RedundancyScheme: customRS},
+				expectedRS: customRS,
+				expectedES: storj.EncryptionScheme{Cipher: defaultES.Cipher, BlockSize: customRS.ShareSize},
+			}, {
+				create:     &storj.CreateObject{EncryptionScheme: customES},
+				expectedRS: defaultRS,
+				expectedES: customES,
+			},
+		} {
+			errTag := fmt.Sprintf("%d. %+v", i, tt)
+
+			obj, err := db.CreateObject(ctx, bucket.Name, TestFile, tt.create)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			info := obj.Info()
+
+			assert.Equal(t, TestBucket, info.Bucket, errTag)
+			assert.Equal(t, TestFile, info.Path, errTag)
+			assert.EqualValues(t, 0, info.Size, errTag)
+			assert.Equal(t, tt.expectedRS.Algorithm, info.Algorithm, errTag)
```

Comment:
Cannot you compare the whole struct at once?

Reply:
Uh, yes... stupid me.